Primera entrega lexer y parser

Liset97 opened the pull request.

Liset97 committed:
Primera entrega lexer y parser

apiad commented:
Pongan el `@handle` de Github de cada miembro del equipo en `doc/Readme`.

Bueno, y arreglar los errores...

Liset97 committed:
cambio

Liset97 committed:
modifica

Liset97 committed:
dos nombres

pablodearmas committed:
Se agregaron las clases para el lexer y el parser
Se agregaron clases para el procesamiento de los errores
Se agregaron algunos tests que faltaban

Liset97 committed:
todos los miembros

pablodearmas committed:
Merge branch 'entrega-parser' of https://github.com/Liset97/cool-compiler-2020 into entrega-parser

pablodearmas committed:
Movimiento de los fuente a las carpeta src

pablodearmas committed:
Se actualizó nombre y versión del compilador

pablodearmas committed:
Se arreglaron varios errores
Se modificaron el lexer y el parser para que sean compatibles con lo que esperan los test automáticos
Se mejoró el código para soportar PyTest en Windows

pablodearmas committed:
Se arreglaron errores en el lexer y el parser para los casos de prueba

apiad commented:
Ayúdenme a entender que es lo que le pasa a su compilador. ¿Y por qué veo cambios en los tests?

pablodearmas committed:
Se arreglo el lexer para detectar caracter nulo

pablodearmas committed:
Se arregló un bug del lexer relativo a las cadenas simples no terminadas correctamente

pablodearmas committed:
Se arreglo un bug de las cadenas simples

pablodearmas committed:
Se arreglo otro bug

pablodearmas committed:
Se arregló el mensaje de error del lexer

pablodearmas committed:
Se arregló un bug relacionado con la diferencia en el EOL entre Windows / Linux

apiad commented:
ENTREGA PARSER ACEPTADA **(CON RETRASO)**
